src/game/laramisc.c

```diff
@@ -576,18 +576,17 @@ void LaraInitialiseMeshes(int32_t level_num)
         g_Lara.mesh_ptrs[i] = g_Meshes[g_Objects[O_LARA].mesh_index + i];
     }
 
-    int16_t gun_type = start->gun_type;
-    if (gun_type == LGT_SHOTGUN) {
-        if (start->flags.got_uzis || start->flags.got_magnums
-            || start->flags.got_pistols) {
-            gun_type = LGT_PISTOLS;
-        } else {
-            gun_type = LGT_UNARMED;
-        }
+    if (g_Lara.gun_status == LGS_ARMLESS) {
+        return;
```

Comment:
You need to move up the shotgun back torso check before this. If Lara has the shotgun in her inventory but not equipped, it doesn't appear on her mesh until she draws it at the start of a new level.

It does load correctly on a saved game, so that must be restoring it from a saved state and not calling this function?

Everything else looks good.

Reply:
Fixed